components/formats-gpl/src/loci/formats/in/BaseZeissReader.java

```diff
@@ -2120,36 +2120,79 @@ public static Charset get(int value) {
-      s += "  text: " + text + "\n";
+      StringBuilder s = new StringBuilder("  SHAPE: ");
+      s.append(id);
+      appendValue(s, "    Type=", type);
+      appendHexValue(s, "    Unknown1=", unknown1);
+      appendHexValue(s, " Unknown2=", unknown2);
+      appendHexValue(s, " Unknown3=", unknown3);
+      s.append("\n    Bbox=");
+      s.append(x1);
+      s.append(",");
+      s.append(y1);
+      s.append("  ");
+      s.append(x2);
+      s.append(",");
+      s.append(y2);
+      appendHexValue(s, "\n    Unknown4=", unknown4);
+      appendHexValue(s, " Unknown5=", unknown5);
+      appendHexValue(s, " Unknown6=", unknown6);
+      appendHexValue(s, " Unknown7=", unknown7);
+      appendHexValue(s, "\n    Unknown8=", unknown8);
+      appendHexValue(s, "\n    Unknown10=", unknown10);
+      appendHexValue(s, " Unknown11=", unknown11);
+      appendHexValue(s, " Unknown12=", unknown12);
+      appendHexValue(s, "\n    Unknown13=", unknown13);
+      appendHexValue(s, " Unknown14=", unknown14);
+      appendHexValue(s, " Unknown15=", unknown15);
+      appendHexValue(s, " Unknown16=", unknown16);
+      appendHexValue(s, "\n    Unknown17=", unknown17);
+      appendHexValue(s, " Unknown18=", unknown18);
+      appendHexValue(s, " Unknown19=", unknown19);
+      appendHexValue(s, "\n    fillColour=", fillColour);
+      appendHexValue(s, " textColour=", textColour);
+      appendHexValue(s, " drawColour=", drawColour);
+      appendValue(s, "\n    drawStyle=", drawStyle);
+      appendValue(s, " fillStyle=", fillStyle);
+      appendValue(s, " lineEndStyle=", lineEndStyle);
+      appendValue(s, " lineEndSize=", lineEndSize);
+      appendValue(s, " lineEndPositions=", lineEndPositions);
+      appendValue(s, "\ndisplayTag=", displayTag);
+      appendValue(s, "\n  fontName=", fontName);
+      appendValue(s, " size=", fontSize);
+      appendValue(s, " weight=", fontWeight);
+      appendValue(s, " bold=", bold);
+      appendValue(s, " italic=", italic);
+      appendValue(s, " underline=", underline);
+      appendValue(s, " strikeout=", strikeout);
+      appendValue(s, " alignment=", textAlignment);
+      appendValue(s, " charset=", charset);
+      appendValue(s, "\n  name: ", name);
+      appendValue(s, "\n  text: ", text);
       //s += "  Zpos: " + zpos;
-      s += "  tagID: " + tagID.getKey() + "\n";
-      s += "  handleSize:" + handleSize + "\n";
-      s += "  pointCount:" + pointCount + "\n    ";
-      for (double point : points)
-        s+=point + ", ";
-      s+= "\n";
-      return s;
+      appendValue(s, "\n  tagID: ", tagID.getKey());
+      appendValue(s, "\n  handleSize:", handleSize);
+      appendValue(s, "\n  pointCount:", pointCount);
+      s.append("\n    ");
+      for (double point : points) {
+        s.append(point);
+        s.append(", ");
+      }
+      s.append("\n");
+      return s.toString();
+    }
+
+    private void appendHexValue(StringBuilder s, String key, int i) {
```

Comment:
looks like these can be `static`

Reply:
The inner class is non-static, so making these methods static resulted in a compile error.

Reply:
Aha, thank you for trying it; sorry for the noise!

Reply:
NB: Longer-term :+1: for making the inner classes static since the reverse could play havoc with at least previous garbage collection implementations.